app/src/components/GrantListWithFilter.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <!-- General filters -->
+  <BaseFilterNav :items="grantRegistryListNav" :button="button" title="grants:" />
+  <GrantList :grants="sortedGrants" :grantMetadata="grantMetadata" />
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType, ref, onUnmounted } from 'vue';
+// --- App Imports ---
+import BaseFilterNav from 'src/components/BaseFilterNav.vue';
+import GrantList from 'src/components/GrantList.vue';
+// --- Store ---
+import useCartStore from 'src/store/cart';
+// --- Types ---
+import { FilterNavButton, FilterNavItem, Grant, GrantMetadataResolution } from '@dgrants/types';
+import { SortingMode } from 'src/types';
+// --- Utils ---
+import { bigNumberComparator } from 'src/utils/sorting';
+
+const defaultSortingMode = 'descending';
+const grantsSortingMode = ref<SortingMode>(defaultSortingMode);
+const shuffleNonce = ref(1);
+
+const filterNavTag = computed(() => {
+  switch (grantsSortingMode.value) {
+    case 'descending': {
+      return 'newest';
+    }
+    case 'ascending': {
+      return 'oldest';
+    }
+    case 'random': {
+      return 'shuffle';
+    }
+    default: {
+      return 'newest';
+    }
+  }
+});
+
+const grantRegistryListNav = computed(
+  () =>
+    <FilterNavItem[]>[
+      {
+        label: 'Sort',
+        tag: filterNavTag.value,
+        menu: [
+          {
+            label: 'newest',
+            action: () => {
+              grantsSortingMode.value = 'descending';
+            },
+          },
+          {
+            label: 'oldest',
+            action: () => {
+              grantsSortingMode.value = 'ascending';
+            },
+          },
+          {
+            label: 'shuffle',
+            action: () => {
+              shuffleNonce.value++;
+              grantsSortingMode.value = 'random';
+            },
+          },
+        ],
+      },
+    ]
+);
+
+export default defineComponent({
+  name: 'GrantListWithFilter',
+  components: { BaseFilterNav, GrantList },
+  props: {
+    // --- Required props ---
+    grants: { type: Array as PropType<Grant[]>, required: true },
+    grantMetadata: { type: Object as PropType<Record<string, GrantMetadataResolution>>, required: true },
+    button: { type: Object as PropType<FilterNavButton>, required: false },
+  },
+  setup(props) {
+    const { addToCart, isInCart, removeFromCart } = useCartStore();
+    const sortedGrants = computed(() => {
+      // Force to recalculate when clicking "Shuffle" consecutively.
+      shuffleNonce.value;
+      return [...props.grants].sort((a, b) => bigNumberComparator(a.id, b.id, grantsSortingMode.value));
+    });
```

Comment:
I think a cleaner way to do this might be to remove the computed property and create a new ref from the `grants` prop which stores the sorted refs. Then have the actions in `grantRegistryListNav` call a method called `sortGrants` which does:

```typescript
const sortedGrants = ref(props.grants);
function sortGrants(order: 'ascending' | 'descending' | 'shuffle') {
   if (order === 'ascending') sortedGrants.value = props.grants.sort((a,b) => a.id > b.id ? 1 : -1)
   else if (order === 'descending') sortedGrants.value = props.grants.sort((a,b) => a.id < b.id ? 1 : -1)
   else sortedGrants.value = props.grants.sort((a,b,) => Math.random() < 0.5 ? 1 ? -1 // shuffle is default
}
```

(I may have the orders wrong but you get the idea). Then you don't need the `bigNumberComparator` and you can remove the shuffle nonce

Reply:
With this change you can also move the `SortingMode` type definition into this file since it's no longer needed anywhere else